### PR TITLE
Express Checkout -  Buyer Pays on PayPal

### PR DESCRIPTION
**Pull request by itbabu**
PayPal gives the possibility to skip the confirm-order page on your website.

My company had a request about this so I had a quick look but then the client was happy with the current implementation.

I'll try to handle this when I have a bit of free time.

In the meantime, if anyone needs this Express Checkout feature this is the place to start.

[documentation here](https://cms.paypal.com/uk/cgi-bin/?cmd=_render-content&content_ID=developer/e_howto_api_ECCustomizing)

> Buyer Pays on PayPal
> With Express Checkout, you can shorten your checkout flow to let buyers complete their purchases on PayPal. Then, you can skip your order confirmation page.
> 
> Generally, buyers select payment methods as the last step before they complete their purchases. If you collect no additional information after buyers return from PayPal, you can skip the confirm-order page on your website. If you collect additional information that does not affect the payment, PayPal recommends that you collect it after buyers complete their purchases.
> 
> The useraction URL parameter in your redirect to PayPal determines whether buyers complete their purchases on PayPal or on your website. If you set useraction to commit, PayPal sets the button text to Pay Now on the PayPal Review your information page. This text lets buyers know that they complete their purchases if they click the button.
> 
> After PayPal redirects buyers to your site, call GetExpressCheckoutDetails and DoExpressCheckoutPayment to have PayPal complete the payment successfully. Call DoExpressCheckoutPayment without waiting for buyer interaction. Use information in the GetExpressCheckoutDetails response to fill out your order confirmation page.

### CODE REVIEW

File: paypal/express/gateway.py [hunk 1]
```diff
@@ -25,12 +25,15 @@
...
 SALE, AUTHORIZATION, ORDER = 'Sale', 'Authorization', 'Order'
 
+
 # The latest version of the PayPal Express API can be found here:
 # https://developer.paypal.com/docs/classic/release-notes/
 API_VERSION = getattr(settings, 'PAYPAL_API_VERSION', '119')
 
 logger = logging.getLogger('paypal.express')
 
+buyer_pays_on_paypal = lambda: getattr(settings, 'PAYPAL_BUYER_PAYS_ON_PAYPAL', False)
```
Comment (itbabu) [inline paypal/express/gateway.py]: Tests: if it's a constant I can't use `override_settings`.
https://code.djangoproject.com/ticket/22071

**Comment by itbabu** [comment]
I don't want to expose the 'Buyer Pays on website' specific urls if I choose 'Buyer Pays on PayPal' option and viceversa.

I tried to use [`reload_url_conf()`](https://github.com/django-oscar/django-oscar/blob/master/tests/functional/checkout/guest_checkout_tests.py#L21) from Oscar without success, so I had to enable all the urls during tests.

**Comment by aleprovencio** [comment]
Any reason why you guys have quit on this issue?

**Comment by itbabu** [comment]
@aleprovencio Hello Alexandre,
Our only client using paypal changed payment gateway so I've almost forgotten this PR :-).

It's waiting for a review.
I think @maikhoepfel is the maintainer now (?).

**Comment by maikhoepfel** [comment]
@aleprovencio @itbabu I'm afraid there's a lack of maintainers for this project at the moment. Neither me nor @codeinthehole really have the time for it. I know @mvantellingen still does Oscar work; maybe he can be convinced. Of course, we're also happy for anybody to step up maintaining it.

**Comment by aleprovencio** [comment]
Still no news guys?

**Comment by SalahAdDin** [comment]
@itbabu What's about this? Any news?

**Comment by woutdp** [comment]
I'd be interested in this feature. The PR is a bit stale, I'd be willing to clean it up if it will be merged.

**Comment by solarissmoke** [comment]
@woutdp I'm happy to review/merge if you are able to clean up the PR.

**Comment by woutdp** [comment]
@solarissmoke 
https://github.com/django-oscar/django-oscar-paypal/pull/193

**Comment by solarissmoke** [comment]
Closing in favour of #193.

**Pull request closed without merging**